datafusion/sql/tests/sql_integration.rs

```diff
@@ -4695,34 +4698,37 @@ fn test_infer_types_from_between_predicate() {
 
     // replace params with values
     let param_values = vec![ScalarValue::Int32(Some(10)), ScalarValue::Int32(Some(30))];
-    let plan_with_params = plan.with_param_values(param_values).unwrap();
-
-    assert_snapshot!(
-        plan_with_params,
-        @r"
-    Projection: person.id, person.age
-      Filter: person.age BETWEEN Int32(10) AND Int32(30)
-        TableScan: person
-    "
-    );
+    let plan_with_params = plan.with_param_values(param_values);
+    match plan_with_params {
+        Ok(plan_p) => assert_snapshot!(
+            plan_p,
+            @r"
+        Projection: person.id, person.age
+          Filter: person.age BETWEEN Int32(10) AND Int32(30)
+            TableScan: person
+        "
+        ),
+        Err(err) => print!("error: {:?}", err),
```

Comment:
If `LogicalPlan::with_param_values` fails, we want this test to fail, right?

Reply:
good catch! I changed this just because I was debugging, this will fail if we do not declare or infer the types in the `prepare`.